src/renderer/graph/ChartController.tsx

```diff
@@ -1,32 +1,45 @@
 import { useMemo } from 'react';
 import * as d3 from 'd3';
+import { Color } from 'renderer/styles/color';
 
 export const GRAPH_START = 10;
 export const GRAPH_END = 25000;
 
-export interface ChartDataPoint {
-  x: number;
-  y: number;
-}
+export const INIT_ANIMATE_DURATION = 750;
+export const GRAPH_ANIMATE_DURATION = 100;
 
-export interface MarginLike {
+export interface IMarginLike {
   left: number;
   top: number;
   right: number;
   bottom: number;
 }
 
-export interface ChartData {
+export interface IChartPointData {
+  x: number;
+  y: number;
+}
+
+export interface IChartLineDataPointsById {
+  [id: string]: IChartPointData[];
+}
+
+export interface IChartCurveData {
+  id: string;
   name: string;
-  color: string; // #ffffff, white, rgba(255, 255, 255, 0.5)
-  items: ChartDataPoint[];
+  line: {
+    color: Color;
+    strokeWidth: number;
```

Comment:
I like the idea of keeping it separate because things like "color" and "strokeWidth" only apply to the "line" part of the curve and not the control point.

Reply:
thats fair.